assets/product-info.js

```diff
@@ -92,33 +85,27 @@ if (!customElements.get('product-info')) {
         productForm?.handleErrorMessage();
       }
 
-      get isFeaturedProduct() {
-        return this.dataset.section.includes('featured_product');
-      }
-
-      handleSwapProduct(productUrl) {
+      handleSwapProduct(productUrl, updateFullPage) {
         return (html) => {
           this.productModal?.remove();
 
-          // Grab the selected variant from the new product info
-          const variant = this.getSelectedVariant(html.querySelector(`product-info[data-section=${this.sectionId}]`));
+          const selector = updateFullPage ? "product-info[id^='MainProduct']" : 'product-info';
+          const variant = this.getSelectedVariant(html.querySelector(selector));
           this.updateURL(productUrl, variant?.id);
 
-          // If we are in an embedded context (quick add, featured product, etc), only swap product info.
-          // Otherwise, refresh the entire page content and sibling sections.
-          if (this.dataset.updateUrl === 'false') {
+          if (updateFullPage) {
```

Comment:
I reversed this conditional so that we can rely on the `updateFullPage` variable instead of re-querying the dom